Create Blog “monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh”

GuopingJia opened the pull request:
Automatically generated by Netlify CMS

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+In [the first blog](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/), we described the detailed process of deploying the complex Online Boutique application to a hybrid cloud environment, across the public EKS cluster from AWS to the private Kubernetes cluster in [HPE GreenLake for Private Cloud Enterprise](https://w ...
+
+In this blog post, we c﻿hoose the Apache SkyWalking as an APM tool and describe how to set up it, as a *self-hosted* APM tool, for monitoring and alerting the application performance across hybrid cloud environment. We take the service mesh as an auto instrumentation mechanism for application monitoring, without adding any manual process to instrument existing applications. 
```
In this blog post, I choose the Apache SkyWalking as an APM tool....

I take the service mesh as an auto instrumentation...

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+---
+## Introduction
+
+In [the first blog](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/), we described the detailed process of deploying the complex Online Boutique application to a hybrid cloud environment, across the public EKS cluster from AWS to the private Kubernetes cluster in [HPE GreenLake for Private Cloud Enterprise](https://w ...
```
In [a previous blog post](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/) I covered the detailed process of deploying....

choukden replied:
Maybe,
[In my previous blog post](https://developer.hpe.com/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh/), I covered the detailed process of deploying....

or

[In my first blog post](https://developer.hpe.com/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh/), I covered the detailed process of deploying....

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apart from a list of supported agents to be used for instrumenting applications, Apache SkyWalking implements an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of *RPC*, this observation  ...
```
Since service mesh provides full control of ***RPC** communication protocol, this observation solution...

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,320 @@
+--set oap.env.SW_ENVOY_METRIC_ALS_TCP_ANALYSIS=mx-mesh
+```
+
+Above Helm commands install the Apache SkyWalking to the namespace *skywalking* of the AWS EKS cluster. It uses the *elasticsearch* as storage type and creates it as *StatefulSet* resource, running a pod on each worker node. It installs the Apache SkyWalking OAP with replicas setting as 2 to ensure the high availability. The installation enables the Envoy Access Log Service (ALS) and specifies th ...
````
It install the Apache SkyWalking Observability Analysis Platform (OAP) with replicas setting....

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+Above Helm commands install the Apache SkyWalking to the namespace *skywalking* of the AWS EKS cluster. It uses the *elasticsearch* as storage type and creates it as *StatefulSet* resource, running a pod on each worker node. It installs the Apache SkyWalking OAP with replicas setting as 2 to ensure the high availability. The installation enables the Envoy Access Log Service (ALS) and specifies th ...
+
+It should be noted that the Apache SkyWalking supports also the ALS analyzer *k8s-mesh*, which uses the metadata from Kubernetes cluster to analyze the logs. It requires the SkyWalking OAP server to access the Kubernetes API server to get information of pods, services and service endpoints. This works only for monitoring a single cloud environment. For hybrid cloud case, we need to use the ALS an ...
```
For hybrid cloud case, you need to use the ALS analyzer ....

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,320 @@
+job.batch/skywalking-oap-init   1/1           31s        94s
+```
+
+We change the service types for both service *skywalking-skywalking-helm-oap*   and *skywalking-skywalking-helm-ui*  from *ClusterIP* to *LoadBalancer*. With the built-in support of Elastic Load Balancing in AWS EKS cluster, the external accessible load balancing host names will be created for those two services in Apache SkyWalking installation.
````
You can change the service type for both.....
With the built-in support of Elastic Load Balancing (ELB) in AWS EKS cluster, ....

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+## Set up Apache SkyWalking
+
+### Prerequisites
```
In the list of prerequisites, you may also want to add the requirement for "istioctl" command line utility installed on the local development environment with hotlink to the istioctl installation: https://istio.io/latest/docs/setup/install/istioctl/

What do you think?

GuopingJia replied:
Good catch ! yes, both of those CLI tools are required. They were in my original content, strange. I've just re-added them.

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+Run above commands on both AWS EKS cluster and private Kubernetes cluster. It installs the *istio* to the namespace *istio-system* of the clusters. It enables explicitly the Envoy Access Log Service (ALS), with both  *envoyAccessLogService.address* and  *envoyMetricsService.address* settings pointing to the Apache SkyWalking OAP server ELB host name and its port *11800*.
+
+We can check the detailed *istio* installation by typing the following *kubectl* command:
```
You can check...

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+### Deploy Online Boutique application
+
+Please follow up the deployment process in [the first blog](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/) to deploy the Online Boutique application across the public AWS EKS cluster and the private Kubernetes cluster in HPE GreenLake for Private Cloud Enterprise. The *Skupper* CLI tool is used for deploying the Vi ...
```
Please follow up the deployment process in my blog post [here](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/) to deploy....

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+The Skupper console shows the connections from the public AWS EKS cluster to the private Kubernetes cluster and the hybrid deployment of the Online Boutique application:
+
+![](/img/skupper-apps.png)
```
The image is not rendered in the preview of the blog post, I believe the reason is that the image name is the same as the image name used in the blog post "How to deploy application across hybrid clouds beyond HPE GreenLake for Private Cloud Enterprise". The image should be rendered once the blog post "How to deploy application across hybrid clouds beyond HPE GreenLake for Private Cloud Enterprise" is published. We will need to check for this when merging the PR to publish your blog posts.

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+The Online Boutique UI can be accessed through deployed frontend service URL from public AWS EKS cluster:
+
+![](/img/online-boutique-frontend.png)
```
The image is not rendered in the preview of the blog post, I believe the reason is that the image name is the same as the image name used in the blog post "How to deploy application across hybrid clouds beyond HPE GreenLake for Private Cloud Enterprise". The image should be rendered once the blog post "How to deploy application across hybrid clouds beyond HPE GreenLake for Private Cloud Enterprise" is published. We will need to check for this when merging the PR to publish your blog posts.

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+
+
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., _service_resp_time_, _ service_instance_resp_time_, and _service_sla_. It will trigger alerting when some metrics reach its pre-defined thresholds. We can define new metrics using SkyWalking observability analysis language (OAL), or customize the existing metrics w ...
```
Please make sure to use escape character (a backslash) in front of "_" character for _service_resp_time and other metrics.

choukden commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,320 @@
+
+
+
+This alert is triggered by the following metric alerting rule for the metric _ service_instance_resp_time _: 
```
Use escape (a backslash) character in front of each "_" for the metric _service_instance_resp_time

choukden reviewed (changes requested):
Hi Guoping,

Some comments and suggestions for edits.

Dale will go through the editorial review in the coming days or so.

Best regards,
Denis

choukden reviewed (changes requested):
Just add one comment about the first paragraph to make a cross reference to the first blog post of the series.

GuopingJia commented:
A new patch has been submitted that should address all your comments.

Guoping

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,331 @@
+```markdown
+$ kukectl exec pod/skywalking-skywalking-helm-oap-bfb57fbf8-5g7k7 -n skywalking -it -- cat /skywalking/config/alarm-settings.yml 
+```
+You can define new metrics by adding new entry to the file using SkyWalking observability analysis language (OAL), or customize the existing metrics with new thresholds. 
````
Add the word "a" prior to "new entry"
Add the word "the" prior to "SkyWalking observability analysis language"

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+
+
+B﻿elow is the alarms page from SkyWalking UI showing all the triggered alerts for deployed Online Boutique application: 
```
Add the word "a" prior to "deployed online"

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,331 @@
+```
+
+
+It indicates issue from the _frontend_ service instance in Online Boutique application. You can check further the service trace page to figure out the root cause for this issue.
````
Add the word "an" prior to "issue from"
For the second sentence, I'd modify it to read:
"You can check **the** service trace page **further** to figure out the root cause **of** this issue."

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+
+
+This blog post discussed the challenges in hybrid cloud monitoring and described the process of setting up Apache SkyWalking as a _self-hosted_ APM tool for application performance monitoring across hybrid cloud environment. Instead of taking the manual instrumentation mechanism to rebuild the application with various agents to collect and send the application metrics, we used an auto instrumenta ...
```
For the end of the first sentence, either add the word "a" prior to "hybrid cloud environment." OR change the word "environment" to its plural: "... across hybrid cloud environments."

For the last sentence, ad the word "an" prior to "application that consists of..."

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+In [my previous blog post](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/), I covered the detailed process of deploying the complex Online Boutique application to a hybrid cloud environment, across the public EKS cluster from AWS to the private Kubernetes cluster in [HPE GreenLake for Private Cloud Enterprise](https ...
+
+In this blog post, I c﻿hoose the Apache SkyWalking as an APM tool and describe how to set up it, as a *self-hosted* APM tool, for monitoring and alerting the application performance across hybrid cloud environment. I take the service mesh as an auto instrumentation mechanism for application monitoring, without adding any manual process to instrument existing applications. 
```
In order to monitor application performance, we normally have to rebuild the application with a kind of *agent* to collect and send metrics data from the application for monitoring purpose. This process refers to as instrumenting an application or the term *instrumentation*.

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apart from a list of supported agents to be used for instrumenting applications, Apache SkyWalking implements an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of _**RPC**_ communication  ...
```
I've rewritten this paragraph to remove the confusion you mentioned. Hope it's clearer now for you.

drensing replied:
Thank you. Yes. It is clearer.

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,331 @@
+job.batch/skywalking-oap-init   1/1           31s        94s
+```
+
+You change the service types for both service *skywalking-skywalking-helm-oap*   and *skywalking-skywalking-helm-ui*  from *ClusterIP* to *LoadBalancer*. With the built-in support of Elastic Load Balancing (ELB) in AWS EKS cluster, the external accessible load balancing host names will be created for those two services in Apache SkyWalking installation.
````
Thank you for your kind explanation. I see it :-)

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+The Skupper console shows the connections from the public AWS EKS cluster to the private Kubernetes cluster and the hybrid deployment of the Online Boutique application:
+
+![](/img/skupper-apps.png)
```
Denis told it could be the same image name being used in both the first blog and this second one. I can double check it when the blog blog gets published. Otherwise, I can re-upload the image with a different name.

drensing replied:
We may need to fix these post publication. I still couldn't see them.

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+The Online Boutique UI can be accessed through deployed frontend service URL from public AWS EKS cluster:
+
+![](/img/online-boutique-frontend.png)
```
could be the same reason.

GuopingJia commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,331 @@
+
+#### 3 Monitor Online Boutique application
+
+After follow up all the steps, the Online Boutique application metrics will be showing up from Apache SkyWalking UI, under *Service Mesh* tab:
```
> Thank you, Guoping. I got the first series of 2 blog posts reviewed. I'll need to wait until later on Monday or early Tuesday to work on the other series.
> 
> All the best, Dale

Thanks Dale for your comments and many good suggestions.

The new patch has addressed all your comments. 

Just let me know if there still anything you want me to change. 

Guoping

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,333 @@
+---
+## Introduction
+
+In [my previous blog post](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/), I covered the detailed process of deploying the complex Online Boutique application in a hybrid cloud environment, across the public EKS cluster from AWS to the private Kubernetes cluster in [HPE GreenLake for Private Cloud Enterprise](https ...
```
This may have been my fault, as I may have suggested an edit that I shouldn't have, but I went back and double-checked.... "on-premise" should be "on-premises", so where it says "It enables the enterprise to rely on the security of on-premise data centers..." you might want to change that (back?) to "on-premises.

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,333 @@
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool that can monitor, trace, and diagnose distributed systems. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apart from a list of supported agents to be used for instrumenting applications, Apache SkyWalking implements an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh under Kubernetes environment, no matter the architecture or language. A service mesh provides a mesh of _Laye ...
```
In the second sentence... "A service mesh provides a mesh of Layer 7 proxies that manages network..." the word "manages" should be "manage".

In the next sentence "service to service" should have hyphens between them --> "service-to-service"

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
```diff
@@ -0,0 +1,333 @@
+### Prerequisites
+
+Before you start, make sure you have the following required elements:
+
```
In these bullets, please remove the semi-colons. You can use a period at the end of each because you have a number of them that have more than one line/sentence in them, like you did in the other blog post in this series. But don't use semi-colons.

drensing commented on content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md:
````diff
@@ -0,0 +1,333 @@
+--set oap.env.SW_ENVOY_METRIC_ALS_TCP_ANALYSIS=mx-mesh
+```
+
+Above the Helm commands install the Apache SkyWalking to the namespace *skywalking* of the AWS EKS cluster. It uses the *elasticsearch* as the storage type and creates it as a *StatefulSet* resource, running a pod on each worker node. It installs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas being set as 2 to ensure high availability. The installation enables the Envoy ...
````
Please place a comma after "Above the Helm commands"

drensing reviewed (commented):
Hi, Guoping. I just had a couple more comments for you to look at on this one prior to publication.

All the best,
Dale

GuopingJia commented:
> Hi, Guoping. I just had a couple more comments for you to look at on this one prior to publication.
> 
> All the best, Dale

Hi, Dale, 

The new patch has addressed your comments. In particular, I've uploaded the images with new names. 
Let me know if you still can't see them in your review windows. 

Guoping
